doc/deployment-schema.json

```diff
@@ -0,0 +1,310 @@
+                                            "required": [
+                                                "type",
+                                                "settings",
+                                                "status",
+                                                "restartPolicy"
+                                            ],
+                                            "properties": {
+                                                "type": {
+                                                    "$ref": "#/definitions/moduleType"
+                                                },
+                                                "settings": {
+                                                    "$ref": "#/definitions/moduleSettings"
+                                                },
+                                                "env": {
+                                                    "$ref": "#/definitions/env"
+                                                },
+                                                "status": {
+                                                    "$ref": "#/definitions/status"
+                                                },
+                                                "restartPolicy": {
+                                                    "$ref": "#/definitions/restartPolicy"
+                                                }
+                                            }
+                                        }
+                                    }
+                                },
+                                "modules": {
+                                    "type": "object",
+                                    "patternProperties": {
+                                        "^[a-zA-Z0-9_.-]*$": {
+                                            "type": "object",
+                                            "required": [
+                                                "type",
+                                                "status",
+                                                "restartPolicy",
+                                                "settings"
+                                            ],
+                                            "properties": {
+                                                "version": {
+                                                    "type": "string",
+                                                    "examples": [
+                                                        "1.0"
+                                                    ]
+                                                },
+                                                "type": {
+                                                    "$ref": "#/definitions/moduleType"
+                                                },
+                                                "status": {
+                                                    "$ref": "#/definitions/status"
```

Comment:
Should it be named as "moduleStatus"?

Reply:
I used moduleType not to confuse with type from schema and moduleSetting because there is a runtime setting, but status is only used for module so I think there is no confusion